### PR TITLE
Test Fixes and Dockerfile Improvements

### PR DESCRIPTION
**Pull request by lingrino**
This PR adds a couple test fixes and rewrites the Dockerfile to build inside the image and then transfer the binary (and tzinfo) to a new scratch image, which should reduce image size and let people build locally without go tooling.

Monkey patching was not working for me until I disabled inlining while running tests (using `-gcflags=-l`). I added that to `make test`. This is recommended in at the bottom of the [Monkey README](https://github.com/bouk/monkey#notes).

It also appears that the error message for an unknown timezone has changed, and I've updated that as well. This may break older go versions (I am `1.13.4`).  Thoughts?

Let me know how this looks, and what's needed to merge. I am also wondering if you'd be open to the following, which I can add here or in a new PR:
- Updating go to the latest version
- Switching to go modules
- Switching to github actions

Thanks! cc @asobti

### CODE REVIEW

**Comment by lingrino** [comment]
Looks like CI is failing with the "cannot find nolnexistent in zip file" vs "unknown time zone nolnexistent" test. Thoughts on how we can support both of these?

File: config/config_test.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func (s *ConfigTestSuite) TestTimezone() {
 
 	// avoid Exit(255) on glog.Fatal
 	monkey.Patch(glog.Fatal, func(a ...interface{}) {
-		s.Contains(a[0], "cannot find nolnexistent in zip file")
+		s.Contains(a[0], "unknown time zone nolnexistent")
```
Comment (asobti) [inline config/config_test.go]:
```go
s.Contains(a[0], "cannot find nolnexistent in zip file") || 
  s.Contains(a[0], "unknown time zone nolnexistent")
```

Comment (asobti) [reply config/config_test.go]: just tried this and this doesn't appear to work. I'm tempted to get rid of this entirely. I dont' think it is worthwhile to test that the program will Fatal exit if an invalid timezone is provided

Comment (lingrino) [reply config/config_test.go]: I agree and just removed those lines from this commit

**Comment by lingrino** [comment]
@asobti I removed those lines and this PR should be ready. Just updates the Dockerfile, adds the `-gcflags=-l` flag for tests and removes that problematic test.

Thanks for https://github.com/asobti/kube-monkey/pull/166 ! I'll follow up with another PR to update to use go modules and the newest go version.

**Comment by cjbehm** [comment]
Would love for this to be merged (obviously, as I inadvertently submitted essentially the same PR heh) and I definitely fought with the test errors when I started out.

**Comment by lingrino** [comment]
I think everything is ready for this to be merged from my side. @asobti let me know if there's anything else to update

**Comment by asobti** [comment]
Thanks for this. merged

**Pull request merged**